src/lender/adapters/deployer.sol

```diff
@@ -60,59 +60,33 @@ contract AdapterDeployer {
     address public mkrEnd;
 
     uint public matBuffer;
+    bool public wired;
 
     constructor(address root_, address clerkFabLike_) {
       root = root_;
       clerkFab = ClerkFabLike(clerkFabLike_);
       mkrDeployer = msg.sender;
     }
 
-    function deployClerk(address currency, address seniorToken) public {
-        require(seniorToken != address(0));
-        clerk = clerkFab.newClerk(currency, seniorToken);
-        AuthLike(clerk).rely(root);
-    }
-
-    function initMKR(address lenderDeployer_, address mkrMgr_, address mkrSpotter_, address mkrVat_, address mkrJug_, address mkrUrn_, address mkrLiq_, address mkrEnd_, uint matBuffer_) public {
-        require(mkrDeployer == msg.sender);
-        lenderDeployer = LenderDeployerLike(lenderDeployer_);
-        mkrMgr = mkrMgr_;
-        mkrSpotter = mkrSpotter_;
-        mkrVat = mkrVat_;
-        mkrJug = mkrJug_;
-        mkrUrn = mkrUrn_;
-        mkrLiq = mkrLiq_;
-        mkrEnd = mkrEnd_;
-        matBuffer = matBuffer_;
-        mkrDeployer = address(1);
-    }
-
-    function deploy(bool wireMgr) public {
-		setupMkrAdapter(wireMgr);
-    }
-
-    function deploy() public {
-        deploy(false);
-    }
+    function deployClerk() public {
+        require(lenderDeployer.seniorToken() != address(0));
```

Comment:
```suggestion
        require(lenderDeployer.clerk() != address(0));
```

Reply:
I guess you mean `require(lenderDeployer.clerk() == address(0));`

Reply:
seniorToken check can stay, as it makes sure that the contract has to be deployed prior to deploying the clerk